### ISSUE TITLE
Visual and experience improvements to existing sub navigation flow

### ISSUE THREAD

**Issue by karmatosed** [WordPress/gutenberg#22087]
Whilst there has been some discussion on a new flow, let's iterate on what we have now and then see where we are.

Currently, this is what you see:

![image](https://user-images.githubusercontent.com/253067/80997985-192f5080-8e3a-11ea-9a8a-f1c6708d3168.png)

I have a few iterations to this that are not that big, but should improve the above.

![2020-05-04 19 09 34](https://user-images.githubusercontent.com/253067/80998426-dcb02480-8e3a-11ea-9fd0-74370df77498.gif)

In this the changes are:
- Only show the + when you are in a sub-navigation. This avoids having more than one at a time.
- Reduce the font size by default of sub-navigation menu.
- Bring back the background on sub-navigation.
- Swap arrow for drop down on the first level of sub-navigation, then for the side on other levels.

I know there are multiple issues, but it felt sensible to create one and get these fixes focused as a starting point.

**Comment by mapk** [comment]
This is a big improvement on the current. Thank you. 

In the prototype above, it looks like you now have to click the arrow to see the next level of subnavs. Is that correct? I like that interaction because it only surfaces that layer of complexity with an intentional click. Right now, they appear when just hovering which feels a bit overwhelming.

The only thing I'm hesitant about is making the subnav font size smaller. I'll default to your expertise here, but I'm uncertain if that's necessary.

**Comment by karmatosed** [comment]
> In the prototype above, it looks like you now have to click the arrow to see the next level of subnavs. Is that correct? I like that interaction because it only surfaces that layer of complexity with an intentional click. Right now, they appear when just hovering which feels a bit overwhelming.

Yes, I sort of feel clicking is better in this case and avoids the interface getting overly complex. We have had both over the course of this block and I feel going back to a click is worth exploring to avoid jumping interfaces.

I could lean towards not doing the font size, it is only down by a step, however, that isn't a deal-breaker for me visually. It's worth noting themes might impact this anyway so we're talking a default experience.

**Comment by adamziel** [comment]
I need to refine it, but an early draft is available here: https://github.com/WordPress/gutenberg/pull/22107

**Comment by karmatosed** [comment]
As there seems some agreement, let's move this too needs dev.

**Comment by adamziel** [comment]
https://github.com/WordPress/gutenberg/pull/22107 is ready for review now

**Comment by draganescu** [comment]
Howdy @karmatosed  in #22107 the interaction is like this:

![appender](https://user-images.githubusercontent.com/107534/81184374-02fbce80-8fb9-11ea-9006-2581782da3fe.gif)

So when a lonely submenu item is clicked the appender is not visible. I know we have the toolbar button but it really feels ... like the appender is suddenly missing. Can't we have a flyout that only contains the appender?

**Comment by adamziel** [comment]
@draganescu @karmatosed The appender change broke multiple tests so I extracted it to a separate PR: 

https://github.com/WordPress/gutenberg/pull/22165

Let's reach a consensus on whether or not we want to land that change before I put in effort to fix those tests.

**Comment by karmatosed** [comment]
@talldan thanks that makes sense, I'll comment over there.

**Comment by noisysocks** [comment]
Once https://github.com/WordPress/gutenberg/pull/22165 is merged into `master` we can close this.

**Comment by draganescu** [comment]
#22165 is merged and closes this issue.
